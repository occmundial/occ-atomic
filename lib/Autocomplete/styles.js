import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
