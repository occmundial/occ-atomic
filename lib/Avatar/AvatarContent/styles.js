import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
