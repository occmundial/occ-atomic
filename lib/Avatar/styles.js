import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
