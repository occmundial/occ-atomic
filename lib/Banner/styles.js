import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
