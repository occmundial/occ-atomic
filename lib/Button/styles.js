import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
