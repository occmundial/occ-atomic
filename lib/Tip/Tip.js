import React from 'react';
import PropTypes from 'prop-types';
import isString from 'lodash/isString';
import flatten from 'lodash/flatten';

import Flexbox from '../Flexbox';
import Text from '../Text';
import Icon from '../Icon';
import colors from '../subatomic/colors';

const boldRegex = /\*(.*?)\*/g;

/** Tip component with different themes (info, warning, success & error). You can pass a string or a react element as the children prop, but the bold regex will only work with strings. */
const Tip = ({ id, style, className, classes, icon, theme, children }) => {
    const replaceStringToBold = () => {
        if (children === '') return children;
        const result = children.split(boldRegex);
        if (result.length === 1) return children;
        for (let i = 1, length = result.length; i < length; i += 2) {
            result[i] = boldMatch(result[i], i, theme);
        }
        return result;
    };

    const boldMatch = (match, i) => {
        return <Text key={i} strong tag="b" {...themeObj}>{match}</Text>;
    };

    const renderText = () => {
        let textArray;
        if (!Array.isArray(children)) textArray = [children];

        return flatten(textArray.map(x => {
            return isString(x) ? replaceStringToBold(x, theme) : x;
        }));
    };

    const getTheme = () => {
        return { [theme]: true };
    };

    const getIconColor = () => {
        return [colors[`${theme}Text`]];
    };

    const themeObj = getTheme(theme);

    return (
        <div id={id} className={className} style={style}>
            <Flexbox display="flex" className={`${classes.container} ${classes[theme]}`}>
                {icon && <Flexbox><Icon iconName={icon} className={classes.icon} colors={getIconColor(theme)} /></Flexbox>}
                <Flexbox flex="1">
                    <Text standard tag="div" {...themeObj}>
                        {renderText()}
                    </Text>
                </Flexbox>
            </Flexbox>
        </div>
    );
};

Tip.defaultProps = {
    theme: 'info'
};

Tip.propTypes = {
    classes: PropTypes.object.isRequired,
    /** Theme of the Tip. */
    theme: PropTypes.oneOf([
        'info', 'warning', 'success', 'error'
    ]),
    /** Text of the tip. If you put text between two * characters that text will be displayed as bold. */
    children: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.node
    ]).isRequired,
    /** Icon name to show on the left. */
    icon: PropTypes.string,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Tip;
