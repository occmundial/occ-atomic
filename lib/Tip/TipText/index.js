import React from 'react';
import PropTypes from 'prop-types';

const boldRegex = /\*(.*?)\*/g;

const TipText = ({ classes, text }) => {
  if (text === '' || typeof text !== 'string') return text;
  return (
    <p className={classes}>
      {text.replace(boldRegex, (_, txt) => txt)}
    </p>
  );
};

TipText.propTypes = {
  classes: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};

export default TipText;
