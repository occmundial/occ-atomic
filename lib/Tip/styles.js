import colors from '../tokens/colors.json';
import fonts from '../tokens/fonts.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import { parseFontValue } from '../Text/styles';

export default {
  container: {
    padding: [spacing['size-4']],
    borderRadius: borderRadius['br-xs'],
    minHeight: '56px',
    alignItems: 'center'
  },
  noBorderRadius: {
    borderRadius: 0
  },
  info: {
    backgroundColor: colors.alert.info.bg,
    outline: `1px solid ${colors.border.info.subtle}`,
    outlineOffset: '-1px'
  },
  warning: {
    backgroundColor: colors.alert.warning.bg,
    outline: `1px solid ${colors.border.warning.subtle}`,
    outlineOffset: '-1px'
  },
  success: {
    backgroundColor: colors.alert.success.bg,
    outline: `1px solid ${colors.border.success.subtle}`,
    outlineOffset: '-1px'
  },
  error: {
    backgroundColor: colors.alert.error.bg,
    outline: `1px solid ${colors.border.error.subtle}`,
    outlineOffset: '-1px'
  },
  promote: {
    backgroundColor: colors.alert.neutral.bg,
  },
  textInfo: {
    color: colors.text.indigo.primary
  },
  textWarning: {
    color: colors.text.warning
  },
  textSuccess: {
    color: colors.text.success
  },
  textError: {
    color: colors.text.error
  },
  textPromote: {
    color: colors.text.white.primary
  },
  icon: {
    marginRight: spacing['size-2'],
    flexShrink: 0,
  },
  text: {
    maxWidth: '100%'
  },
  closeIcon: {
    marginLeft: spacing['size-4']
  },
  cta: {
    textDecoration: 'underline',
    font: parseFontValue(fonts['alert-default']),
    cursor: 'pointer',
    textWrap: 'nowrap',
  },
  ctaLarge: {
    marginLeft: spacing['size-4'],
    alignSelf: 'center',
  },
  ctaSmall: {
    margin: [spacing['size-2'], 0, 0],
    alignSelf: 'start',
  },
  normalText: {
    font: parseFontValue(fonts['alert-default']),
    display: 'inline-block',
		margin: 0
  },
  textContainer: {
    padding: [spacing['size-3'], 0]
  },
  small: {
    maxWidth: 360,
    padding: spacing['size-3'],
  }
};
