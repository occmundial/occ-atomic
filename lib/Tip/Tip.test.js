import React from 'react';
import { shallow, mount } from 'enzyme';
import Tip from './Tip';
import styles from './styles';
import TipJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Tip", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Tip classes={classes}>Testing</Tip>
        );
        expect(wrapper).toMatchSnapshot();
    });

	it('has the right theme', () => {
		const wrapper = shallow(
            <Tip classes={classes} theme="info">Testing</Tip>
        );
        expect(wrapper.find('.info').length).toBe(1);
		expect(wrapper.find('.warning').length).toBe(0);
        wrapper.setProps({ theme: 'warning' });
        expect(wrapper.find('.info').length).toBe(0);
		expect(wrapper.find('.warning').length).toBe(1);
	});

    it('has an icon', () => {
        const wrapper = shallow(
            <Tip classes={classes} icon>Alert text</Tip>
        );
        expect(wrapper.find('.icon').length).toBe(1);
    });

    it('has a close icon button', () => {
        const wrapper = mount(
            <Tip 
                classes={classes}
                icon 
                onClose={() => console.log('Close')}
                testId="alert-test"
            >Tip text </Tip>
        );
        expect(wrapper.find({ "data-testid": 'alert-test__close-icon'}).length).toBe(1);
    });

    it('has a link', () => {
        const wrapper = mount(
            <Tip 
                classes={classes}
                icon 
                cta={{
                    text: 'Click here.',
                    onClick: () => alert('CTA clicked!')
                }}
                testId="alert-test"
            >Tip text </Tip>
        );
        expect(wrapper.find({ "data-testid": 'alert-test__link'}).length).toBe(1);
    });

});

describe("TipJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <TipJSS>Tip text </TipJSS>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Tip styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});
