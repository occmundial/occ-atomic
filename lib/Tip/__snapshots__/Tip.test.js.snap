// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tip matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tip
    classes={
      Object {
        "container": "container",
        "error": "error",
        "icon": "icon",
        "info": "info",
        "noBorderRadius": "noBorderRadius",
        "spacedClose": "spacedClose",
        "success": "success",
        "text": "text",
        "textWithIcon": "textWithIcon",
        "warning": "warning",
      }
    }
    theme="info"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Jss(Flexbox)
        className="container info"
        display="flex"
        justifyContent="start"
      >
        <Jss(Flexbox)
          alignItems="center"
          display="flex"
          flex="1"
          justifyContent={null}
        >
          <div
            className="text"
          >
            <Jss(Text)
              info={true}
              standard={true}
              tag="div"
            />
          </div>
        </Jss(Flexbox)>
      </Jss(Flexbox)>,
      "className": undefined,
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Jss(Flexbox)
            alignItems="center"
            display="flex"
            flex="1"
            justifyContent={null}
          >
            <div
              className="text"
            >
              <Jss(Text)
                info={true}
                standard={true}
                tag="div"
              />
            </div>
          </Jss(Flexbox)>,
          undefined,
        ],
        "className": "container info",
        "display": "flex",
        "justifyContent": "start",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "center",
            "children": Array [
              undefined,
              <div
                className="text"
              >
                <Jss(Text)
                  info={true}
                  standard={true}
                  tag="div"
                />
              </div>,
            ],
            "display": "flex",
            "flex": "1",
            "justifyContent": null,
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Jss(Text)
                  info={true}
                  standard={true}
                  tag="div"
                />,
                "className": "text",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    undefined,
                  ],
                  "info": true,
                  "standard": true,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Jss(Flexbox)
          className="container info"
          display="flex"
          justifyContent="start"
        >
          <Jss(Flexbox)
            alignItems="center"
            display="flex"
            flex="1"
            justifyContent={null}
          >
            <div
              className="text"
            >
              <Jss(Text)
                info={true}
                standard={true}
                tag="div"
              />
            </div>
          </Jss(Flexbox)>
        </Jss(Flexbox)>,
        "className": undefined,
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Jss(Flexbox)
              alignItems="center"
              display="flex"
              flex="1"
              justifyContent={null}
            >
              <div
                className="text"
              >
                <Jss(Text)
                  info={true}
                  standard={true}
                  tag="div"
                />
              </div>
            </Jss(Flexbox)>,
            undefined,
          ],
          "className": "container info",
          "display": "flex",
          "justifyContent": "start",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alignItems": "center",
              "children": Array [
                undefined,
                <div
                  className="text"
                >
                  <Jss(Text)
                    info={true}
                    standard={true}
                    tag="div"
                  />
                </div>,
              ],
              "display": "flex",
              "flex": "1",
              "justifyContent": null,
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Jss(Text)
                    info={true}
                    standard={true}
                    tag="div"
                  />,
                  "className": "text",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      undefined,
                    ],
                    "info": true,
                    "standard": true,
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Tip styles matches the snapshot 1`] = `
Object {
  "container": Object {
    "alignItems": "center",
    "borderRadius": 6,
    "padding": 16,
  },
  "error": Object {
    "backgroundColor": "rgba(219, 55, 55, 0.1)",
  },
  "icon": Object {
    "marginRight": 8,
  },
  "info": Object {
    "backgroundColor": "rgba(87, 54, 171, 0.1)",
  },
  "noBorderRadius": Object {
    "borderRadius": 0,
  },
  "spacedClose": Object {
    "marginRight": 48,
  },
  "success": Object {
    "backgroundColor": "rgba(0, 149, 55, 0.1)",
  },
  "text": Object {
    "maxWidth": "100%",
  },
  "textWithIcon": Object {
    "maxWidth": "calc(100% - 32px)",
  },
  "warning": Object {
    "backgroundColor": "rgba(255, 197, 73, 0.1)",
  },
}
`;

exports[`TipJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Tip)
    theme="info"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container": "Tip-container-0-1-73",
        "error": "Tip-error-0-1-78",
        "icon": "Tip-icon-0-1-79",
        "info": "Tip-info-0-1-75",
        "noBorderRadius": "Tip-noBorderRadius-0-1-74",
        "spacedClose": "Tip-spacedClose-0-1-82",
        "success": "Tip-success-0-1-77",
        "text": "Tip-text-0-1-80",
        "textWithIcon": "Tip-textWithIcon-0-1-81",
        "warning": "Tip-warning-0-1-76",
      },
      "theme": "info",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container": "Tip-container-0-1-73",
          "error": "Tip-error-0-1-78",
          "icon": "Tip-icon-0-1-79",
          "info": "Tip-info-0-1-75",
          "noBorderRadius": "Tip-noBorderRadius-0-1-74",
          "spacedClose": "Tip-spacedClose-0-1-82",
          "success": "Tip-success-0-1-77",
          "text": "Tip-text-0-1-80",
          "textWithIcon": "Tip-textWithIcon-0-1-81",
          "warning": "Tip-warning-0-1-76",
        },
        "theme": "info",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
